src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -480,7 +480,8 @@ protected override void OnItemsChanged(object e)
 		{
 			if (e is IVectorChangedEventArgs iVCE)
 			{
-				if (iVCE.CollectionChange == CollectionChange.ItemChanged || iVCE.CollectionChange == CollectionChange.ItemInserted)
+				if (iVCE.CollectionChange == CollectionChange.ItemChanged
+				    || (iVCE.CollectionChange == CollectionChange.ItemInserted && iVCE.Index < Items.Count))
```

Comment:
Why did you need this - how can OnItemsChanged() be called with an inserted item that's out of range?

Reply:
It was happening during my tests: sometimes the index was out of range on the items in the event args and this was crashing the app.